junit-platform-testkit/src/main/java/org/junit/platform/testkit/engine/Events.java

```diff
@@ -336,4 +398,60 @@ private static void assertEventsMatchExactly(List<Event> events, Condition<? sup
 		softly.assertAll();
 	}
 
+	@SafeVarargs
+	private static void assertEventsMatchLoosely(List<Event> events, Condition<? super Event>... conditions) {
+		SoftAssertions softly = new SoftAssertions();
+		for (Condition<? super Event> condition : conditions)
+			checkCondition(events, softly, condition);
+		softly.assertAll();
+	}
+
+	@SafeVarargs
+	@SuppressWarnings("varargs")
+	private static void assertEventsMatchIncompleteButOrdered(List<Event> events,
+			Condition<? super Event>... conditions) {
+		Assertions.assertThat(conditions).hasSizeLessThanOrEqualTo(events.size());
+		SoftAssertions softly = new SoftAssertions();
+
+		// @formatter:off
+		List<Integer> indices = Arrays.stream(conditions)
+				.map(condition -> findEvent(events, softly, condition))
+				.filter(Objects::nonNull)
+				.map(events::indexOf)
+				.collect(toList());
+		// @formatter:on
+
+		if (isNotInIncreasingOrder(indices))
+			softly.fail("Conditions are not in the correct order.");
```

Comment:
For this particular use case, does `softly.fail(..)` provide benefit over immediately throwing an `AssertionFailedError`?

Reply:
probably not - taking a look.